src/Form/index.tsx

```diff
@@ -1058,7 +1065,9 @@ function Form({
       submitData || ['button', 'text'].includes(metadata.elementType);
     if (!redirectKey) {
       if (explicitNav) {
+        setFormCompleted(true);
```

Comment:
let's remove this boolean state and instead directly use the session boolean variable?